homeassistant/core.py

```diff
@@ -165,9 +161,10 @@ def async_start(self):
 
         # pylint: disable=protected-access
         self.loop._thread_ident = threading.get_ident()
-        _async_create_timer(self)
         self.bus.async_fire(EVENT_HOMEASSISTANT_START)
+        yield from self.async_stop_track_tasks()
```

Comment:
So this fixes the fact that `CoreState.starting` was never detectable before, as we would immediately overwrite it. 

However, this has the downside that it is impossible for components/platforms to start a permanent async job as a result from EVENT_HOMEASSISTANT_START.